src/prefect/events/clients.py

```diff
@@ -195,3 +241,173 @@ async def _emit(self, event: Event) -> None:
                     # a standard load balancer timeout, but after that, just take a
                     # beat to let things come back around.
                     await asyncio.sleep(1)
+
+
+class PrefectCloudEventSubscriber:
+    """
+    Subscribes to a Prefect Cloud event stream, yielding events as they occur.
+
+    Example:
+
+        from prefect.events.clients import PrefectCloudEventSubscriber
+        from prefect.events.filters import EventFilter, EventNameFilter
+
+        filter = EventFilter(event=EventNameFilter(prefix=["prefect.flow-run."]))
+
+        async with PrefectCloudEventSubscriber(api_url, api_key, filter) as subscriber:
+            async for event in subscriber:
+                print(event.occurred, event.resource.id, event.event)
+
+    """
+
+    _websocket: Optional[WebSocketClientProtocol]
+    _filter: "EventFilter"
+    _seen_events: "Mapping[UUID, bool]"
+
+    def __init__(
+        self,
+        api_url: str = None,
+        api_key: str = None,
+        filter: "EventFilter" = None,
+        reconnection_attempts: int = 10,
+    ):
+        """
+        Args:
+            api_url: The base URL for a Prefect Cloud workspace
+            api_key: The API of an actor with the manage_events scope
+            reconnection_attempts: When the client is disconnected, how many times
+                the client should attempt to reconnect
+        """
+        api_url, api_key = _get_api_url_and_key(api_url, api_key)
+
+        from prefect.events.filters import EventFilter
+
+        self._filter = filter or EventFilter()
+
+        try:
+            self._seen_events = TTLCache(maxsize=500_000, ttl=120)
```

Comment:
I know we're using `cachetools` in other places, but I didn't see it in our requirements.txt. It looks like it's getting pulled in as a dependency of a dependency ( `kubernetes` -> `google-auth` -> `cachetools`). What do you say to just packing it into our own dependencies?

Reply:
Ha, okay, `cachetools` adds a whopping 136K to the install size, I'll include it